api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -78,4 +79,69 @@ public PatientQueue getIncompletePatientQueue(Patient patient, Location location
 		return dao.getIncompletePatientQueue(patient, locationTo);
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getMostRecentQueue(org.openmrs.Patient)
+	 */
+	@Override
+	public PatientQueue getMostRecentQueue(Patient patient) {
+		return dao.getMostRecentQueue(patient);
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#assignVisitNumber(org.openmrs.module.patientqueueing.model.PatientQueue)
+	 */
+	@Override
+	public PatientQueue assignVisitNumber(PatientQueue patientQueue) {
+		Date today = new Date();
+		List<PatientQueue> patientQueueList = getPatientQueueList(null, OpenmrsUtil.firstSecondOfDay(today),
+		    OpenmrsUtil.getLastMomentOfDay(today), null, null, patientQueue.getPatient(), null);
+		
+		if (!patientQueueList.isEmpty() && patientQueueList.get(0).getVisitNumber() != null) {
+			patientQueue.setVisitNumber(patientQueueList.get(0).getVisitNumber());
+		} else {
+			patientQueue.setVisitNumber(generateVisitNumber(patientQueue.getLocationFrom(), patientQueue.getPatient()));
+		}
+		return patientQueue;
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#generateVisitNumber(org.openmrs.Location,
+	 *      org.openmrs.Patient)
+	 */
+	public String generateVisitNumber(Location location, Patient patient) {
+		
+		Date today = new Date();
+		
+		SimpleDateFormat formatterExt = new SimpleDateFormat(Context.getAdministrationService().getGlobalProperty(
+		    "patientqueueing.defaultDateFormat"));
+		
+		List<PatientQueue> patientQueues = getPatientQueueList(null, OpenmrsUtil.firstSecondOfDay(today),
```

Comment:
Should this just call getPatientQueueForToday?

Reply:
@dkayiwa no. If it does it won't be getting the latest visit number on a day at a given location. It will end up getting the most recent for the patient